packages/dds/tree/src/test/simple-tree/discrepancies.spec.ts

```diff
@@ -0,0 +1,909 @@
+					mismatch: "nodeKind",
+					view: MapNodeStoredSchema,
+					stored: undefined,
+				},
+				{
+					identifier: schemaFactory.string.identifier,
+					mismatch: "nodeKind",
+					view: undefined,
+					stored: LeafNodeStoredSchema,
+				},
+				{
+					identifier: "schema discrepancies.tree2",
+					mismatch: "nodeKind",
+					view: undefined,
+					stored: ObjectNodeStoredSchema,
+				},
+			],
+		);
+	});
+
+	it("Differing fields on object node schema", () => {
+		// Both utilize ObjectNodeSchema but with differing fieldSchemas
+		const objectNodeSchema1 = schemaFactory.objectAlpha(testTreeNodeIdentifier, {
+			x: schemaFactory.required(schemaFactory.number),
+		});
+		const objectNodeRoot1 = schemaFactory.optional(objectNodeSchema1);
+
+		const objectNodeSchema2 = schemaFactory.objectAlpha(testTreeNodeIdentifier, {
+			x: schemaFactory.optional(schemaFactory.string),
+			y: schemaFactory.optional(schemaFactory.string),
+		});
+		const objectNodeRoot2 = schemaFactory.optional(objectNodeSchema2);
+		const objectNodeStoredSchema2 = toStoredSchema(objectNodeRoot2);
+
+		assert.deepEqual(
+			Array.from(getAllowedContentDiscrepancies(objectNodeRoot1, objectNodeStoredSchema2)),
+			[
+				{
+					identifier: schemaFactory.number.identifier,
+					mismatch: "nodeKind",
+					view: LeafNodeStoredSchema,
+					stored: undefined,
+				},
+				{
+					identifier: testTreeNodeIdentifierNormalized,
+					mismatch: "fields",
+					differences: [
+						{
+							fieldKey: "x",
+							identifier: testTreeNodeIdentifierNormalized,
+							mismatch: "allowedTypes",
+							view: [{ metadata: {}, type: schemaFactory.number }],
+							stored: [schemaFactory.string.identifier],
+						},
+						{
+							fieldKey: "x",
+							identifier: testTreeNodeIdentifierNormalized,
+							mismatch: "fieldKind",
+							view: "Value",
+							stored: "Optional",
+						},
+						{
+							fieldKey: "y",
+							identifier: testTreeNodeIdentifierNormalized,
+							mismatch: "fieldKind",
+							view: "Forbidden",
+							stored: "Optional",
+						},
+					],
+				},
+				{
+					identifier: schemaFactory.string.identifier,
+					mismatch: "nodeKind",
+					view: undefined,
+					stored: LeafNodeStoredSchema,
+				},
+			],
+		);
+	});
+
+	const createLeafNodeSchema = (
+		leafValue: ValueSchema,
+		treeName: string,
+		root: TreeFieldStoredSchema,
+	): TreeStoredSchema => ({
+		rootFieldSchema: root,
+		nodeSchema: new Map([
+			[brand<TreeNodeSchemaIdentifier>(treeName), new LeafNodeStoredSchema(leafValue)],
+		]),
+	});
+
+	it("Differing value types on leaf node schema", () => {
+		const leafNodeSchema1 = schemaFactory.optional(
+			new LeafNodeSchema(testTreeNodeIdentifier, ValueSchema.Number),
+		);
+
+		const root = fieldSchema(FieldKinds.optional, [testTreeNodeIdentifier]);
+
+		const leafNodeSchema2 = createLeafNodeSchema(
+			ValueSchema.Boolean,
+			testTreeNodeIdentifier,
+			root,
+		);
+		const leafNodeSchema3 = createLeafNodeSchema(
```

Comment:
These tests are a bit hard to follow when both the view and stored schema re following the same naming conventions.

Its extra odd with leafNodeSchema1 sounding like a node schema, but being a view FieldSchema and leafNodeSchema2 being a stored tree schema.

Instead of using 1, 2, 3 to differentiate, can we call the stored ones "stored". Something like `optionalNumber` , storedBooleanRoot` and `storedNumberRoot` would be way easier to follow.